webpack/assets/javascripts/react_app/components/Head/index.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+
+const Header = props => (
+  <Helmet>
+    {Object.entries(props).map(([TagName, value]) => (
```

Comment:
Instead of running over all the props, would it make sense to use a dedicated prop so it can be documented using prop-types?

Reply:
Why not just do
```js
<Helmet>{props.children}</Helmet>
```
?

On that note, why is the `<Header>` component necessary at all?  Why not just use `<Helmet>` directly?  The [usage docs](https://github.com/nfl/react-helmet#example) outline a pretty straightforward usage; I don't see a reason to complicate it like this.  @johnpmitsch do you happen to know?

(I plan to raise this concern in Katello as well)

Reply:
The [changelog](https://github.com/nfl/react-helmet/blob/master/CHANGELOG.md#500-2017-03-21) implies that previous versions of `react-helmet` took lots of props like this, but that is being deprecated.  I guess that may answer my question above.  But since we are already on v5+ and migrating to 6, I think my suggestion above still stands.

Reply:
Don't react helmet children need to be actual jsx components? We are passing an object like `{ title: 'foo' }` and the map converts it to `<Title>Foo</Title>`.

If we want to keep the existing functionality but expand it to be able to pass in custom tags, you could add an extra `children` prop value passed in like this (untested):
```js
const Header = props => {
  const { children, ...tags } = props;
  <Helmet>
    {Object.entries(tags).map(([TagName, value]) => (
      <TagName key={TagName}>{value}</TagName>
    ))}
    {children && (children)};
  </Helmet>
);
```

which would allow you to pass something like this `{ title: 'foo', children: <meta charset="UTF-8" /> }` to `withHeaders`

But I would also support moving to just passing in actual components as children and updating the usage.

I wonder what we need to support though? Isn't a lot of the head metadata handled by rails for Foreman + plugins? If we are only using it to change the displayed title, I don't see why we need to change.

Reply:
I don't think we need any of this.  (by the way, React will take care of `{props.children}` whether or not it's blank, and there is no need to specify it as an explicit prop.)  I think the existing map is a holdover from the previous react-helmet api which is no longer needed.

Reply:
@jeremylenz  I was following the Katello code, where everything gets passed to the `withHeaders` function, for example https://github.com/Katello/katello/blob/0e9ae146969244b6f1150dda854a10d07074c144/webpack/containers/Application/config.js#L20

With the loop, we are still passing children, just dynamically.

Mind giving an example? I'm not clear on what the changes to the existing code would look like for your suggestion.

Reply:
I understand a bit better now.  Looks like the complexity is coming from our `withHeader` higher-order component:
```js
function withHeader(WrappedComponent, tags) {
  const Headers = props => (
    <React.Fragment>
      <Header {...tags} />
      <WrappedComponent {...props} />
    </React.Fragment>
  );

  return Headers;
}
```

Because `withHeader` is an HOC and not a regular component, `tags` is passed as a JS object as a second argument, and not as JSX children as one would do otherwise.  So the `tags` must be processed in our Header component like so:

```js
const Header = props => (
  <Helmet>
    {Object.entries(props).map(([TagName, value]) =>
      <TagName key={TagName}>{value}</TagName>)};
  </Helmet>
);
```

However, if you look at our _use_ of the Header component itself, we're only using it to pass in a `title`:

```js
// webpack/containers/Application/config.js
export const links = [
  {
    path: 'redhat_repositories',
    component: WithOrganization(withHeader(Repos, { title: __('RH Repos') })),
  },
  {
    path: 'subscriptions',
    component: WithOrganization(withHeader(Subscriptions, { title: __('Subscriptions') })),
  },
  {
    path: 'subscriptions/add',
    component: WithOrganization(withHeader(UpstreamSubscriptions, { title: __('Add Subscriptions') })),
  },
...
```

So we could change the Header component to this:

```js
const Header = props => (
  <Helmet>
    {props.children}
  </Helmet>
);
```

Now, we can change `withHeader` to this:
```js
function withHeader(WrappedComponent, title) {
  const Headers = props => (
    <React.Fragment>
      <Header>
         <title>{title}</title>
      </Header>
      <WrappedComponent {...props} />
    </React.Fragment>
  );

  return Headers;
}
```

But we're beginning to see here that the Header component really isn't adding much value.  We could just as soon do the following:

1. Delete the Header component
2.  Use `withHeader` with `<Helmet>` directly:

```js
function withHeader(WrappedComponent, title) {
  const Headers = props => (
    <React.Fragment>
      <Helmet>
         <title>{title}</title>
      </Helmet>
      <WrappedComponent {...props} />
    </React.Fragment>
  );

  return Headers;
}
```